Opacity Slider For New Layer Picker

arottersman opened the pull request:
# Overview
Adds the opacity slider to the layer picker from #1666 

It will only appear for coverage layers.

# Demo
#### Wireframe
![image](https://cloud.githubusercontent.com/assets/7633670/22082680/b06278fe-dd96-11e6-9568-e54b97c315b6.png)

#### Firefox
<img width="332" alt="screen shot 2017-01-18 at 3 52 42 pm" src="https://cloud.githubusercontent.com/assets/7633670/22082541/3632ade2-dd96-11e6-9145-866777eb2fe5.png">

#### Chrome
<img width="400" alt="screen shot 2017-01-18 at 3 46 53 pm" src="https://cloud.githubusercontent.com/assets/7633670/22082553/40d8236c-dd96-11e6-8f7e-cc6e0bf9e08b.png">

#### IE 11

![screen shot 2017-01-18 at 3 48 28 pm](https://cloud.githubusercontent.com/assets/7633670/22082557/459cf864-dd96-11e6-92b1-7eea1eea4047.png)

😮 
<img width="400" alt="screen shot 2017-01-18 at 3 48 50 pm" src="https://cloud.githubusercontent.com/assets/7633670/22082559/475a8202-dd96-11e6-835e-f022106549cd.png">

# Testing Instructions
- Pull branch and rebundle
- Confirm other layers work as before and coverage grid layers display a slider in the heading.
  -  The slider should re-render to its start position on each selection of a new layer, and disappear when all the coverage grid layers are off, or when switching to a non-coverage layer tab

Connects #1652 
Connects #1677

arottersman commented:
No screenshot for safari because of https://github.com/WikiWatershed/model-my-watershed/issues/1668

arottersman commented:
@azavea-bot rebuild

rajadain commented:
One behavioral issue is because the opacity slider scrolls out of view when selecting a layer further down, I have to scroll back up to change the opacity:

![2017-01-25 11 41 45](https://cloud.githubusercontent.com/assets/1430060/22299787/6cb07252-e2f3-11e6-9143-f2e8bf572591.gif)

Would it be possible to make the top row (which has the title **Coverage Grid** as well as the opacity slider) sticky?

Any opinions on this @jfrankl?

jfrankl commented:
I agree, seeing it in action, it should be sticky. Alice, let me know how
you want to handle this. Happy to pair on it, happy to let you try to do
it, happy to take it on myself.

On Wed, Jan 25, 2017 at 11:45 AM, Terence Tuhinanshu <
notifications@github.com> wrote:

> One behavioral issue is because the opacity slider scrolls out of view
> when selecting a layer further down, I have to scroll back up to change the
> opacity:
>
> [image: 2017-01-25 11 41 45]
> <https://cloud.githubusercontent.com/assets/1430060/22299787/6cb07252-e2f3-11e6-9143-f2e8bf572591.gif>
>
> Would it be possible to make the top row (which has the title *Coverage
> Grid* as well as the opacity slider) sticky?
>
> Any opinions on this @jfrankl <https://github.com/jfrankl>?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/WikiWatershed/model-my-watershed/pull/1667#issuecomment-275161393>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABud9LSUc6iMhOKPvxn2Cnq8yh0ai9tfks5rV3wVgaJpZM4LnUuZ>
> .
>



-- 
Jeff Frankl, User Experience Designer

Azavea  |  990 Spring Garden Street, 5th Floor, Philadelphia, PA
Web azavea.com  |  Blog azavea.com/blogs  |  Twitter @azavea
<http://twitter.com>

arottersman commented:
I can get the stickiness all set up, and I'll grab you (@jfrankl) for the related tweaks (I'm thinking we'll want to adjust the height)

arottersman commented:
Added a styling commit:
![styled_layerpicker](https://cloud.githubusercontent.com/assets/7633670/22302325/3b0b98cc-e2fc-11e6-9fb0-fa4d3162e9de.gif)

Also addressed #1677 (layerpicker visible on login)
<img width="960" alt="screen shot 2017-01-25 at 12 44 46 pm" src="https://cloud.githubusercontent.com/assets/7633670/22302329/3e76fde4-e2fc-11e6-8804-072971738af1.png">

rajadain commented:
Since we're doing styling now, can we also make the cursor into a hand / finger / link when hovering over the tab buttons?

![image](https://cloud.githubusercontent.com/assets/1430060/22303417/37775fde-e301-11e6-9545-d36bccc4a9ee.png)

It isn't obvious that the stars are clickable if the cursor doesn't change.

arottersman commented:
Good point - the fixup makes it 👆  now.

rajadain commented:
The old opacity slider had a larger _hit area_, so you could click around the slider and still get it to change:

![2017-01-25 13 38 48](https://cloud.githubusercontent.com/assets/1430060/22304122/2a0befba-e304-11e6-9af2-699aee6dd3c5.gif)

But the new slider is pretty narrow, and I have to click _exactly_ on the thin strip to get it to change:

![2017-01-25 13 39 25](https://cloud.githubusercontent.com/assets/1430060/22304143/3f91ad70-e304-11e6-8398-bbea91021c4e.gif)

arottersman commented:
There's the same issue on bees - I wasn't able to increase the hit area without messing up the slider styling. I can give fixing it another go unless @jfrankl thinks he can work his magic.

rajadain commented:
We can split that off into its own issue then.

rajadain commented:
Consider adding the following to `.layerpicker` styling:

```scss
.layerpicker {
  border-radius: 2px;
  box-shadow: 0 0 6px $black-74;
}
```

Which brings it in line with the rest, and makes it slightly easier to differentiate from the basemap:

![image](https://cloud.githubusercontent.com/assets/1430060/22304664/65ab1bfc-e306-11e6-8c0d-655691447711.png)

arottersman commented:
- increased click target for slider (will make an issue on bees)
- added drop shadow 
<img width="350" alt="screen shot 2017-01-25 at 2 07 03 pm" src="https://cloud.githubusercontent.com/assets/7633670/22305007/e2519a7c-e307-11e6-8324-6a21fbf70b21.png">

![screen shot 2017-01-25 at 2 07 16 pm](https://cloud.githubusercontent.com/assets/7633670/22305008/e2522190-e307-11e6-93e2-0dae4b407850.png)

rajadain commented:
On Chrome on Windows, the scrollbars eat into the layer name space, pushing the legend icon below

![image](https://cloud.githubusercontent.com/assets/1430060/22304985/cea2ed14-e307-11e6-89cc-84598de309ee.png)

Not sure what the best solution there is. Perhaps limiting how wide the name can be so that there's always room for the legend icon on the right?

rajadain commented:
There is excessive whitespace around the opacity slider on IE11

![image](https://cloud.githubusercontent.com/assets/1430060/22305371/3e3cef5c-e309-11e6-8f1a-676176bcee20.png)

If the slider is not visible, the dialog looks okay:

![image](https://cloud.githubusercontent.com/assets/1430060/22305386/4e78eb6e-e309-11e6-9d1e-96a3bad37d18.png)

rajadain commented:
On Firefox and IE11 on Windows, the legend icons are in the next line and on the left even in cases when the name isn't long:

![image](https://cloud.githubusercontent.com/assets/1430060/22305480/b41b7f90-e309-11e6-8437-4787f8560cba.png)

Chrome on Windows behaves correctly:

![image](https://cloud.githubusercontent.com/assets/1430060/22305492/bf304faa-e309-11e6-87bf-3719dda14642.png)

rajadain commented:
What is the fifth star supposed to show? I get a JavaScript error when I click it

![image](https://cloud.githubusercontent.com/assets/1430060/22305643/4254c94c-e30a-11e6-8ea1-c8af32909202.png)

![image](https://cloud.githubusercontent.com/assets/1430060/22305627/37890dca-e30a-11e6-9551-c84755a32737.png)

arottersman commented:
Thanks for reviewing! The last commit should mostly fix it all up on windows:

#### IE11 slider (still some whitespace, but not such an egregious amount 😉 )
![screen shot 2017-01-25 at 3 41 45 pm](https://cloud.githubusercontent.com/assets/7633670/22308884/e44525d8-e316-11e6-9af0-a9c8f5007178.png)

#### Windows Firefox one line layer names
![screen shot 2017-01-25 at 3 53 36 pm](https://cloud.githubusercontent.com/assets/7633670/22308881/e21706aa-e316-11e6-89ab-1da030edb1e8.png)

#### IE11 one line layer names
![screen shot 2017-01-25 at 3 53 44 pm](https://cloud.githubusercontent.com/assets/7633670/22308882/e221a042-e316-11e6-96d5-89c2a4512ca2.png)

rajadain commented:
Taking a final look at this.

rajadain commented:
@azavea-bot rebuild
